apps/arena/native/physics/src/collision_detection/sat.rs

```diff
@@ -0,0 +1,279 @@
+        // the axis will be the perpendicular line drawn from the current line of the polygon
+        axis = Position {
+            x: -current_line.y,
+            y: current_line.x,
+        };
+
+        // FIXME normalizing on this loop may be bad
+        axis.normalize();
+        let (min_polygon_cast_point, max_polygon_cast_point) =
+            project_vertices(&polygon.vertices, axis);
+        let (min_circle_cast_point, max_circle_cast_point) = project_circle(circle, axis);
+
+        // If there's a gap between the polygon it means they do not collide and we can safely return false
+        if min_polygon_cast_point >= max_circle_cast_point
+            || min_circle_cast_point >= max_polygon_cast_point
+        {
+            return (false, normal, result_depth);
+        }
+
+        let circle_overlap_depth = max_circle_cast_point - min_polygon_cast_point;
+
+        let polygon_overlap_depth = max_polygon_cast_point - min_circle_cast_point;
+
+        let min_depth = f32::min(circle_overlap_depth, polygon_overlap_depth);
+
+        if min_depth < result_depth {
+            // If we hit the polygon from the right or top we need to turn around the direction
+            if polygon_overlap_depth > circle_overlap_depth {
+                normal = Position {
+                    x: -axis.x,
+                    y: -axis.y,
+                };
+            } else {
+                normal = axis;
+            }
+            result_depth = min_depth;
+        }
+    }
+
+    // Check normal and depth for center
+    let closest_vertex = find_closest_vertex(&circle.position, &polygon.vertices);
+    axis = Position::sub(closest_vertex, circle.position);
+    axis.normalize();
+
+    let (min_polygon_cast_point, max_polygon_cast_point) =
+        project_vertices(&polygon.vertices, axis);
+    let (min_circle_cast_point, max_circle_cast_point) = project_circle(circle, axis);
+
+    // If there's a gap between the polygon it means they do not collide and we can safely return false
+    if min_polygon_cast_point >= max_circle_cast_point
+        || min_circle_cast_point >= max_polygon_cast_point
+    {
+        return (false, normal, result_depth);
+    }
+
+    let circle_overlap_depth = max_circle_cast_point - min_polygon_cast_point;
+
+    let polygon_overlap_depth = max_polygon_cast_point - min_circle_cast_point;
+
+    let axis_depth = f32::min(circle_overlap_depth, polygon_overlap_depth);
+
+    if axis_depth < result_depth {
+        // If we hit the polygon from the right or top we need to turn around the direction
+        if polygon_overlap_depth > circle_overlap_depth {
+            normal = Position {
+                x: -axis.x,
+                y: -axis.y,
+            };
+        } else {
+            normal = axis;
+        }
+        result_depth = axis_depth;
+    }
+
+    (true, normal, result_depth)
+}
+
+// Uncomment this if we need a polygon-polygon collision detection
```

Comment:
This was something that i added as an extra since we can calculate this with this algorithm, we can fix it when we need polygon-polygon collisions (if we ever need them)